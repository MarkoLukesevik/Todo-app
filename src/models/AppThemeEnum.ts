export enum AppThemeEnum {
    Light = 0,
    Dark = 1
}
