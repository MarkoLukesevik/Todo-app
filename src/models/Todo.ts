export interface Todo {
  id: string
  title: string
  isComplete: false
}
