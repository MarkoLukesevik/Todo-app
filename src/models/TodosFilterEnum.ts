export enum TodosFilterEnum {
    All = 0,
    Active = 1,
    Completed = 2
}