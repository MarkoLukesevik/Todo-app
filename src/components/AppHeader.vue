<script setup lang="ts">
import { computed } from 'vue'
import { AppThemeEnum } from '@/models/AppThemeEnum'

const props = defineProps<{
  appTheme: AppThemeEnum
}>()

const emit = defineEmits<{
  (e: 'toggleTheme', value: AppThemeEnum): void
}>()

const themeIcon = computed<string>(() => {
  return props.appTheme === AppThemeEnum.Light
    ? '/assets/images/icon-moon.svg'
    : '/assets/images/icon-sun.svg'
})

const handleIconClick = (): void => {
  props.appTheme === AppThemeEnum.Light
    ? emit('toggleTheme', AppThemeEnum.Dark)
    : emit('toggleTheme', AppThemeEnum.Light)
}
</script>

<template>
  <div class="app-header">
    <h1 class="app-header__title">TODO</h1>
    <img class="app-header__theme-icon" :src="themeIcon" alt="" @click="handleIconClick" />
  </div>
</template>

<style scoped lang="scss">
.app-header {
  display: flex;
  flex-direction: row;
  flex: 0 1 auto;
  align-items: center;
  justify-content: space-between;

  color: #ffffff;

  font-family: 'Josefin Sans', sans-serif;

  &__title {
    font-size: 2rem;
  }
}
</style>
