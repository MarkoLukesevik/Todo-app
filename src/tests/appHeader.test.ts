import { describe, it, expect, beforeEach } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import { nextTick } from 'vue';

import { AppThemeEnum } from '@/models/AppThemeEnum';

import AppHeader from '@/components/AppHeader.vue';

describe('Test suite for app header component', () => {
    let wrapper: any;

    beforeEach(() => {
        wrapper = shallowMount(AppHeader, {
            props: { appTheme: AppThemeEnum.Light }
        });
    });

    it('Should match the snapshot', () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the correct theme icon based on the appTheme prop', async () => {
        expect(wrapper.find('[data-test="app-header-image"]').attributes('src')).toBe('/assets/images/icon-moon.svg');

        await wrapper.setProps({ appTheme: AppThemeEnum.Dark });
        await nextTick();

        expect(wrapper.find('[data-test="app-header-image"]').attributes('src')).toBe('/assets/images/icon-sun.svg');
    });

    it('emits toggleTheme with the correct value when the icon is clicked', async () => {
        const apHeaderImage = await wrapper.find('[data-test="app-header-image"]');
        await apHeaderImage.trigger('click');
        expect(wrapper.emitted('toggleTheme')[0]).toEqual([AppThemeEnum.Dark]);

        await wrapper.setProps({ appTheme: AppThemeEnum.Dark });
        await apHeaderImage.trigger('click');
        expect(wrapper.emitted('toggleTheme')[1]).toEqual([AppThemeEnum.Light]);
    });
});