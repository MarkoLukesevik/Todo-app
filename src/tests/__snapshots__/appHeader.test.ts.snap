// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test suite for app header component > Should match the snapshot 1`] = `
<div
  class="app-header"
  data-v-48a13218=""
>
  <h1
    class="app-header__title"
    data-v-48a13218=""
  >
    TODO
  </h1>
  <img
    alt=""
    class="app-header__theme-icon"
    data-test="app-header-image"
    data-v-48a13218=""
    src="/assets/images/icon-moon.svg"
  />
</div>
`;
